pygam/tests/test_GAM_methods.py

```diff
@@ -327,4 +328,85 @@ def test_get_params_hidden():
     params = gam.get_params(deep=True)
     assert('_lam' in list(params.keys()))
 
+
+class TestSamplingFromPosterior(object):
+
+    def test_drawing_samples_from_unfitted_model(self, mcycle, mcycle_gam):
+        X, y = mcycle
+        gam = LinearGAM()
+
+        with pytest.raises(AttributeError):
+            gam.sample(X, y)
+
+        with pytest.raises(AttributeError):
+            gam._sample_coef(X, y)
+
+        with pytest.raises(AttributeError):
+            gam._bootstrap_samples_of_smoothing(X, y)
+
+        assert mcycle_gam._is_fitted
+
+        mcycle_gam.sample(X, y, n_draws=2)
+        mcycle_gam._sample_coef(X, y, n_draws=2)
+        mcycle_gam._bootstrap_samples_of_smoothing(X, y, n_bootstraps=1)
+        assert True
+
+    def test_sample_quantity(self, mcycle, mcycle_gam):
+        X, y = mcycle
+        for quantity in ['coefficients', 'response']:
+            with pytest.raises(ValueError):
```

Comment:
cool. i've never seen this pattern before.
does this tell pytest to expect a ValueError?

Reply:
Yep! Here's the [documentation](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions). An example from there:
```
import pytest

def test_zero_division():
    with pytest.raises(ZeroDivisionError):
        1 / 0
```